quickwit-index/src/actors/source/file_source.rs

```diff
@@ -106,11 +119,22 @@ impl AsyncActor for FileSource {
                 reached_eof = true;
                 break;
             }
-            raw_doc_batch.docs.push(doc_line);
-            self.file_position.num_bytes += num_bytes as u64;
-            self.file_position.line_num += 1u64;
+            docs.push(doc_line);
+            self.counters.current_offset += num_bytes as u64;
+            self.counters.num_lines_processed += 1;
         }
-        if !raw_doc_batch.docs.is_empty() {
+        if !docs.is_empty() {
+            let mut checkpoint_delta = CheckpointDelta::default();
+            checkpoint_delta.add_partition(
```

Comment:
so currently we merge all arriving messages into one one index?
Where would the distributor be to group partitions. On the sender side, as an actor step or here by having multiple batches?

Reply:
One source actor can possibly work on several partitions.
Several source can end up in a single index, they just need to work on different partitions.

That's how we will handle indexing distribution.

The coordination (deciding how many source are indexing, which partitions belongs to whom, and where to spawn them) will be the role of one leader, materialized by an actor on a single server.